spec/types/jtd-schema.spec.ts

```diff
@@ -0,0 +1,212 @@
+/* eslint-disable @typescript-eslint/no-empty-interface */
+import type {JTDSchema} from "../.."
+import chai from "../chai"
+const should = chai.should()
+
+describe("JTDSchema typechecks", () => {
+  it("should typecheck number schemas", () => {
+    const numf: JTDSchema<number> = {type: "float64"}
+    should.exist(numf)
+    const numi: JTDSchema<number> = {type: "int32"}
+    should.exist(numi)
+    // @ts-expect-error
+    const numl: JTDSchema<number> = {type: "int64"}
+    should.exist(numl)
+  })
+
+  it("should typecheck boolean schemas", () => {
+    const bool: JTDSchema<boolean> = {type: "boolean"}
+    should.exist(bool)
+    // boolean literals can't be reduced
+    const boolTrue: JTDSchema<true> = {type: "boolean"}
+    should.exist(boolTrue)
+  })
+
+  it("should typecheck string schemas", () => {
+    const str: JTDSchema<string> = {type: "string"}
+    should.exist(str)
+    const time: JTDSchema<string> = {type: "timestamp"}
+    should.exist(time)
+  })
+
+  it("should typecheck enumeration schemas", () => {
+    const enumerate: JTDSchema<"a" | "b"> = {enum: ["a", "b"]}
+    should.exist(enumerate)
+    // don't need to specify everything
+    const enumerateMissing: JTDSchema<"a" | "b" | "c"> = {enum: ["a", "b"]}
+    should.exist(enumerateMissing)
+    // @ts-expect-error
+    const enumerateNumber: JTDSchema<"a" | "b" | 5> = {enum: ["a", "b"]}
+    should.exist(enumerateNumber)
+    // @ts-expect-error
+    const enumerateString: JTDSchema<"a" | "b"> = {type: "string"}
+    should.exist(enumerateString)
+  })
+
+  it("should typecheck elements schemas", () => {
+    const elements: JTDSchema<number[]> = {elements: {type: "float64"}}
+    should.exist(elements)
+    // homogenous tuples works
+    const tupleHomo: JTDSchema<[number, number]> = {elements: {type: "float64"}}
+    should.exist(tupleHomo)
+    // not heterogeneous
+    const tupleHeteroNum: JTDSchema<[number, string]> = {
+      // @ts-expect-error
+      elements: {type: "float64"},
+    }
+    should.exist(tupleHeteroNum)
+    const tupleHeteroString: JTDSchema<[number, string]> = {
+      // @ts-expect-error
+      elements: {type: "string"},
+    }
+    should.exist(tupleHeteroString)
+  })
+
+  it("should typecheck values schemas", () => {
+    const values: JTDSchema<Record<string, number>> = {values: {type: "float64"}}
+    should.exist(values)
+    const valuesDefined: JTDSchema<{prop: number}> = {values: {type: "float64"}}
+    should.exist(valuesDefined)
```

Comment:
As I commented above, I think in this case "values" should not be allowed - not sure if it is possible though.